components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java

```diff
@@ -62,10 +62,12 @@ RemoteFetchConfiguration getRemoteFetchConfiguration(String fetchConfigurationId
     /**
      * This method is used to call by clients to get list of BasicRemoteFetchConfiguration by tenantID.
      * This method is called by clients in list view.
+     * @param limit
+     * @param offset
      * @return List of BasicRemoteFetchConfiguration
      * @throws RemoteFetchCoreException
      */
-    List<BasicRemoteFetchConfiguration> getBasicRemoteFetchConfigurationList()
+    List<BasicRemoteFetchConfiguration> getBasicRemoteFetchConfigurationList(Integer limit, Integer offset)
             throws RemoteFetchCoreException;
```

Comment:
What's the reason to use `Integer` instead of `int`

Reply:
limit and offset are not mandatory values in the request. It may be null (if URL has no query param as limit). Then we have to load default value from identity.xml. To check the null, used Integer instead of int.

Reply:
We need to use Optional container if it indeed optional.
Using box type and null check is not very optimal and may cause NPE somewhere.

https://docs.oracle.com/javase/8/docs/api/java/util/OptionalInt.html

Reply:
ACK. From the swagger definition, we can only have "limit" and "offset" as  Integer objects.
If we use int in OSGi service param 
- Anyway, we need to handle NPE in REST API (caller)
- Need to send 2 flags to say limit and offset are null or not, so that OSGi service can load from indentity.xml if query param not given.